embed rest-explorer into binary

tbg opened the pull request:
this avoids having to have the `ui` directory in the correct
location relative to the cockroach binary.

renamed the `resources` directory to `resource` so that we can have the
package `resource` provide the embedded resources.

petermattis commented:
LGTM. I had it on my TODO list to investigate doing something like this. Glad you got to it first.

spencerkimball commented:
Is anyone worried about what will happen when we really expand the
resources directory to include all of the management functionality we're
going to build?

On Monday, March 9, 2015, Peter Mattis <notifications@github.com> wrote:

> LGTM. I had it on my TODO list to investigate doing something like this.
> Glad you got to it first.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/394#issuecomment-77852272>.
>

petermattis commented:
I'm not particularly worried. Godoc does something similar (https://github.com/golang/tools/tree/master/godoc/static) and it has a much larger set of static files. If the size of the embedded stuff eventually gets too large there are alternate approaches that can be taken then.

bdarnell commented on resource/resource.go:
```diff
@@ -0,0 +1,27 @@
+// When changing files included here, you may add the `-debug` flag, which will
+// avoid embedding any files (but instead passes them through from disk). This
+// avoids having to recompile before testing changes.
+//go:generate go-bindata -pkg resource -o ./embedded.go ./ui/...
```
`go-bindata` should be added to the glockfile as well as a `cmd` (add the `cmd` line at the top of the file and then glock save to record its version).

tschottdorf replied:
done.

bdarnell commented:
LGTM. 

One thing to watch out for is that the compiler can require huge amounts of memory when compiling large generated files in race or coverage mode, but we can deal with that when/if it becomes a problem.

tbg committed:
embed rest-explorer into binary

this avoids having to have the `ui` directory in the correct
location relative to the cockroach binary.

renamed the `resources` directory to `resource` so that we can have the
package `resource` provide the embedded resources.
